MFVNeutralino/test/K0Tracks/hists.cc

```diff
@@ -121,53 +235,141 @@ int main(int argc, char** argv) {
+	  //muon_iso = muons.iso(i);
+	  //muon_nsigmadxybs = muons.nsigmadxybs(i, bs);
+	  tmpz_p4 += muon_p4;
+
+	}
+	if (nselmuons == 2 && muon_q * muons.q(i) == -1) {
+	  TLorentzVector antimuon_p4;
+	  antimuon_p4.SetPxPyPzE(muons.px(i), muons.py(i), muons.pz(i), muons.p(i));
+	  tmpz_p4 += antimuon_p4;
+	  //has_Zmumuboson = true;
+          if (tmpz_p4.M() > 80 && tmpz_p4.M() < 100) {has_Zmumuboson = true;} //Alec added this line and commented above line so invariant mass consistent with Z boson
+	}
+      }
+    }
+
+    //double z_m = -99;
+    //double zmumu_m = -99; 
+    //double z_pT = -99;
+    //double zmumu_pT = -99;
+    if (has_Zmumuboson) {
+      //zmumu_m = tmpz_p4.M();
+      //zmumu_pT = tmpz_p4.Pt();
+      zmumu_p4 = tmpz_p4;
+    }
+
+    /*int nseleles = 0;
+    //double ele_pT = -99;
+    double ele_p = -99;
+    double ele_q = -99;
+    double ele_px = -99;
+    double ele_py = -99;
+    double ele_pz = -99;
+    //double ele_abseta = -99;
+    //double ele_iso = 99;
+    //double ele_absdxybs = -99;
+    //double ele_absdz = -99;
+    //double ele_nsigmadxybs = -99;
+    TLorentzVector ele_p4;
+    tmpz_p4.SetPxPyPzE(0.0, 0.0, 0.0, 0.0);
+
+    for (int i = 0, ie = electrons.n(); i < ie; ++i) {
+      double tmp_ele_abseta = abs(electrons.eta(i));
+      double tmp_ele_absdxybs = abs(electrons.dxybs(i, bs));
+      double tmp_ele_absdz = electrons.dzpv(i, pvs);  
+      //abs((electrons.vz(i) - pvs.z(0)) - ((electrons.vx(i) - pvs.x(0)) * electrons.px(i) + (electrons.vy(i) - pvs.y(0)) * electrons.py(i)) / electrons.pt(i) * electrons.pz(i) / electrons.pt(i));
+      bool ele_IP_cut = tmp_ele_abseta < 1.48 ? tmp_ele_absdxybs < 0.05 && tmp_ele_absdz < 0.1 : tmp_ele_absdxybs < 0.1 && tmp_ele_absdz < 0.2;
+      if (ele_IP_cut && electrons.pt(i) > 20.0 && abs(electrons.eta(i)) < 2.4 && electrons.isTight(i) && electrons.passveto(i) && electrons.iso(i) < 0.1) {
+	nseleles += 1;
+	if (nseleles == 1) {
+	  //ele_pT = electrons.pt(i);
+	  ele_p = electrons.p(i);
+	  ele_px = electrons.px(i);
+	  ele_py = electrons.py(i);
+	  ele_pz = electrons.pz(i);
+	  //ele_abseta = abs(electrons.eta(i));
+	  //ele_absdxybs = abs(electrons.dxybs(i, bs));
+	  //ele_absdz = electrons.dzpv(i, pvs);
+	  //abs((electrons.vz(i) - pvs.z(0)) - ((electrons.vx(i) - pvs.x(0)) * electrons.px(i) + (electrons.vy(i) - pvs.y(0)) * electrons.py(i)) / electrons.pt(i) * electrons.pz(i) / electrons.pt(i));
+	  ele_p4.SetPxPyPzE(ele_px, ele_py, ele_pz, ele_p);
+	  ele_q = electrons.q(i);
+	  //ele_iso = electrons.iso(i);
+	  //ele_nsigmadxybs = electrons.nsigmadxybs(i, bs);
+	  tmpz_p4 += ele_p4;
+
+	}
+	
+	if (nseleles > 0 && ele_q * electrons.q(i) == -1) {
+	  TLorentzVector antiele_p4;
+	  antiele_p4.SetPxPyPzE(electrons.px(i), electrons.py(i), electrons.pz(i), electrons.p(i));
+	  tmpz_p4 += antiele_p4;
+	  //has_Zeeboson = true;
+	}
+      }
+    }*/  //Alec to this line
+
```

Comment:
@duquale There are a lot of things in this that are commented out--are they things that nobody would ever need, or are they being saved in case we wanted to go back to some alternative options you had studied at some point or another? Just trying to think of ways to streamline and simplify this, if possible. If these are alternatives that could possibly come in handy some day, it might be okay to leave in (maybe with a comment mentioning what it would be for).